src/app/CommandHandler.h

```diff
@@ -134,6 +199,7 @@ class CommandHandler : public Command
     Callback * mpCallback = nullptr;
     InvokeResponseMessage::Builder mInvokeResponseBuilder;
     TLV::TLVType mDataElementContainerType = TLV::kTLVType_NotSpecified;
+    size_t mRefCount                       = 0;
```

Comment:
Can we use `ReferenceCounted` in `src/lib/core/ReferenceCounted.h` here ? Instead of implementing `IncRef`/`DecRef`. We can move the logic inside `DecRef` into the destructor, or the deletor.